src/core/border_router/routing_manager.cpp

```diff
@@ -72,6 +72,7 @@ RoutingManager::RoutingManager(Instance &aInstance)
     , mDiscoveredPrefixInvalidTimer(aInstance, HandleDiscoveredPrefixInvalidTimer)
     , mDiscoveredPrefixStaleTimer(aInstance, HandleDiscoveredPrefixStaleTimer)
     , mRouterAdvertisementCount(0)
+    , mLastRouterAdvertisementSendTime(TimerMilli::GetNow())
```

Comment:
Just noticed this should be set to `TimerMilli::GetDistantPast()`. @simonlingoogle

(This may be why we need to wait additional durations in `tests/scripts/thread-cert/border_router/test_multi_thread_networks.py`

Reply:
I am not sure about using this here.  The `DistantPast` and `DistantFuture` only make sense and meaningful  relative to a specific point of time (they are tied to some specific time instance) and we need to use them when dealing with same `GetNow()` value.

To give an example, if `GetNow()` is `0x1` now. And we calculate `GetNow().GetDistantPast()`, this is with respect to `0x01` value. If time moves forward and `GetNow()` is now `0x5` then the calculated previous distant past is no longer valid (it may be even considered a future time now with respect to `0x5` value).

Reply:
I think using `TimerMilli::GetNow() - kMinDelayBetweenRtrAdvs` is the best solution. 
I will file a new PR to do that.  @wgtdkp

Reply:
Created PR: https://github.com/openthread/openthread/pull/7489

PTAL. @wgtdkp @abtink